### PR TITLE
Update VEP runner to use Life Sciences API beta & VEP 104

### LINKED ISSUE #650: Update `vep_runner.py` to use Life Sciences `v2beta` API
Right now the [pipelines_runner.sh](https://github.com/googlegenomics/gcp-variant-transforms/blob/master/docker/pipelines_runner.sh) uses the `v2beta` (Life Sciences) API while the [vep_runner.py](https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/libs/annotation/vep/vep_runner.py) uses the `v2alpha1` (Genomics) API.

This isn't a huge deal, but in a new project it did mean enabling both APIs instead of just one. For people new to gcp-variant-transforms, this adds a level of complexity, especially if something goes wrong that needs to be debugged.

On a related note, you might consider using [dsub](https://github.com/DataBiosphere/dsub) in the vep_runner instead of calling the Pipelines API directly. `dsub` has built-in support for preemptible VMs and retries at the individual task (shard) level. It could make running the annotations pipeline much less expensive if preemptible VMs were used. Bashir left a note to add this in the code:

```
                    # TODO(bashir2): Add the option of using preemptible
                    # machines and the retry functionality.
                    'preemptible': False,
 ```

`dsub` supports both the `v2alpha1` and `v2beta` APIs. Happy to help if you are interested.

### PR DESCRIPTION
**Pull request by moschetti**
- Update vep_runner & tests to use Cloud LifeSciences API, including adding `--location` flag
- Update defaults to use newest VEP release (104) (Underlying GCS & GCR updates in: https://github.com/googlegenomics/variant-annotation/pull/15)

I will re-run tests after the variant-annotation updates and the new container push.

Fixes #650, #655, #665

### CODE REVIEW

**Comment by hnawar** [comment]
@moschetti Please note this merge was blocked and it is much needed.
Thanks

**Comment by hnawar** [comment]
Also it seems the ftp path, reverted to small vep instead of caps
ftp://ftp.ensembl.org/pub/release-104/variation/vep/homo_sapiens_vep_104_GRCh38.tar.gz

File: gcp_variant_transforms/libs/annotation/vep/vep_runner_test.py [hunk 1]
```diff
@@ -103,19 +112,19 @@ def test_make_vep_cache_path(self):
...
-        self._mock_service, 'mouse', 'mm9', _INPUT_PATTERN, _OUTPUT_DIR,
-        _VEP_INFO_FIELD, _IMAGE, '', _NUM_FORK, self._get_pipeline_args(),
-        None, 30)
+        self._mock_service, _LOCATION, 'mouse', 'mm9', _INPUT_PATTERN,
+        _OUTPUT_DIR, _VEP_INFO_FIELD, _IMAGE, '', _NUM_FORK,
+        self._get_pipeline_args(), None, 30)
     self.assertEqual(test_instance._vep_cache_path,
                      ('gs://cloud-lifesciences/vep/'
-                      'vep_cache_mouse_mm9_91.tar.gz'))
+                      'vep_cache_mouse_mm9_104.tar.gz'))
```
Comment (pgrosu) [inline gcp_variant_transforms/libs/annotation/vep/vep_runner_test.py:127]: @moschetti There does not exist a `mouse_mm9`, though there is a `mus_musculus`:

```Bash
[pgrosu]$ gsutil ls gs://cloud-lifesciences/vep/
gs://cloud-lifesciences/vep/
gs://cloud-lifesciences/vep/vep_cache_homo_sapiens_GRCh37_104.tar.gz
gs://cloud-lifesciences/vep/vep_cache_homo_sapiens_GRCh37_91.tar.gz
gs://cloud-lifesciences/vep/vep_cache_homo_sapiens_GRCh38_104.tar.gz
gs://cloud-lifesciences/vep/vep_cache_homo_sapiens_GRCh38_91.tar.gz
gs://cloud-lifesciences/vep/vep_cache_mus_musculus_GRCm39_104.tar.gz
[pgrosu]$
```

**Review by pgrosu — COMMENTED** [review]
@moschetti There does not exist a `mouse_mm9`, though there is a `mus_musculus`:

```Bash
[pgrosu]$ gsutil ls gs://cloud-lifesciences/vep/
gs://cloud-lifesciences/vep/
gs://cloud-lifesciences/vep/vep_cache_homo_sapiens_GRCh37_104.tar.gz
gs://cloud-lifesciences/vep/vep_cache_homo_sapiens_GRCh37_91.tar.gz
gs://cloud-lifesciences/vep/vep_cache_homo_sapiens_GRCh38_104.tar.gz
gs://cloud-lifesciences/vep/vep_cache_homo_sapiens_GRCh38_91.tar.gz
gs://cloud-lifesciences/vep/vep_cache_mus_musculus_GRCm39_104.tar.gz
[pgrosu]$
```

File: gcp_variant_transforms/libs/annotation/vep/vep_runner_test.py (hunk 1 above)
Comment (moschetti) [reply gcp_variant_transforms/libs/annotation/vep/vep_runner_test.py]: Thanks for catching that. Updated in new commit.

**Comment by moschetti** [comment]
@hnawar we're working on as couple other fixes as well right now with a plan to cut a new release shortly with multiple fixes so this is on hold to merge until then.

**Pull request merged**